cdc/split.cc

```diff
@@ -370,156 +378,102 @@ api::timestamp_type find_timestamp(const mutation& m) {
     return v._ts;
 }
 
-bool should_split(const mutation& base_mutation) {
-    const auto& base_schema = *base_mutation.schema();
-    auto& p = base_mutation.partition();
+struct should_split_visitor {
+    bool _had_static_row = false;
+    bool _had_clustered_row = false;
+    bool _had_upsert = false;
+    bool _had_row_marker = false;
+    bool _had_range_delete = false;
+
+    bool _result = false;
+
+    api::timestamp_type _ts = api::missing_timestamp;
+    std::optional<gc_clock::duration> _ttl = std::nullopt;
 
-    api::timestamp_type found_ts = api::missing_timestamp;
-    std::optional<gc_clock::duration> found_ttl; // 0 = "no ttl"
+    inline bool finished() const { return _result; }
+    inline void stop() { _result = true; }
 
-    auto check_or_set = [&] (api::timestamp_type ts, gc_clock::duration ttl) {
-        if (found_ts != api::missing_timestamp && found_ts != ts) {
-            return true;
+    void visit(api::timestamp_type ts, gc_clock::duration ttl = gc_clock::duration(0)) {
```

Comment:
While I am completely throwing stones in glass houses, this logic is one of the many places I think could use some brief comments sprinkled on "why this means this and that"...

Reply:
By "this logic", I mean the whole visitor...

Reply:
I agree, I'll add comments.

Reply:
Added comments, check'em out